bson/mgocompat/registry.go

```diff
@@ -82,3 +85,28 @@ func newRegistryBuilder() *bsoncodec.RegistryBuilder {
 
 	return rb
 }
+
+// buildMgoRNVRegistry creates a new bsoncodec.RegistryBuilder configured to behave like mgo/bson with RespectNilValues set to true.
+func buildMgoRNVRegistry() *bsoncodec.RegistryBuilder {
```

Comment:
If this is building a registry, should it return a `bson.Registry` instead of a `bsoncodec.RegistryBuilder`? The same comment applies to `buildMgoRegistry`. Given these are unexported functions, I'm also ok deferring this change until we do the ticket to export the new registries.

Also, instead of including `RNV` in the name, let's just expand it to avoid acronym confusion and call it `buildMgoRespectNilValuesRegistry`.

Reply:
done